src/screens/DiscoveryScreenV2/elements/DiscoveryItemList.js

```diff
@@ -25,19 +25,71 @@ const DomainList = (props) => {
     isHashtag,
     handleSetFollow,
     handleSetUnFollow,
+    handleSetBlock,
+    handleSetUnblock,
     onPressBody,
     DefaultImage,
-    isCommunity
+    isCommunity,
+    isBlockedSection
   } = props;
 
+  const renderButonAction = () => {
+    if (isBlockedSection) {
+      if (item.isUnblocked) {
+        return (
+          <TouchableNativeFeedback testID="isUnblock" onPress={handleSetBlock}>
+            <View style={styles.followContainer}>
+              <View style={styles.buttonBlockUser}>
+                <Text style={styles.textButtonBlockUser}>Block</Text>
+              </View>
+            </View>
+          </TouchableNativeFeedback>
+        );
+      }
+      return (
+        <TouchableNativeFeedback testID="isBlock" onPress={handleSetUnblock}>
+          <View style={styles.followContainer}>
+            <View style={styles.buttonBlock}>
+              <Text style={styles.textButtonBlock}>Blocked</Text>
+            </View>
+          </View>
+        </TouchableNativeFeedback>
+      );
+    }
+    if (item.isunfollowed) {
+      return (
+        <TouchableNativeFeedback onPress={handleSetFollow}>
+          <View style={styles.followContainer}>
+            <View style={styles.buttonFollow}>
+              <Text style={styles.textButtonFollow}>{isCommunity ? 'Join' : 'Follow'}</Text>
+            </View>
+          </View>
+        </TouchableNativeFeedback>
+      );
+    }
+    return (
+      <TouchableNativeFeedback onPress={handleSetUnFollow}>
+        <View style={styles.followContainer}>
+          <View style={styles.buttonFollowing}>
+            <Text style={styles.textButtonFollowing}>{isCommunity ? 'Joined' : 'Following'}</Text>
+          </View>
+        </View>
+      </TouchableNativeFeedback>
+    );
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `renderButtonAction` function has been refactored to handle different button actions based on conditions. However, it seems that there is no error handling or fallback case if none of the conditions are met. It would be better to add a default return statement at the end of this function to handle such cases.
```diff
  const renderButonAction = () => {
    ...
    return (
      <TouchableNativeFeedback onPress={handleSetUnFollow}>
        ...
      </TouchableNativeFeedback>
    );
+   return null;
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->